instrumentation/sql/vendor_mysql.go

```diff
@@ -0,0 +1,142 @@
+package sql
+
+import (
+	"net"
+	"strings"
+)
+
+type mysqlExtension struct{}
+
+func init() {
+	vendorExtensions = append(vendorExtensions, &mysqlExtension{})
+}
+
+// Gets if the extension is compatible with the component name
+func (ext *mysqlExtension) IsCompatible(componentName string) bool {
+	return componentName == "mysql.MySQLDriver"
+}
+
+// Complete the missing driver data from the connection string
+func (ext *mysqlExtension) ProcessConnectionString(connectionString string, configuration *driverConfiguration) {
+	configuration.peerService = "mysql"
+
+	dsn := *ext.parseDSN(connectionString)
+	configuration.user = dsn["User"]
+	configuration.port = dsn["Port"]
+	configuration.instance = dsn["DBName"]
+	configuration.host = dsn["Host"]
+	configuration.connString = strings.ReplaceAll(connectionString, dsn["Passwd"], "******")
+}
+
+// ParseDSN parses the DSN string to a Config
+func (ext *mysqlExtension) parseDSN(dsn string) *values {
+	// New config with some default values
+	tmpCfg := values{}
+
+	// [user[:password]@][net[(addr)]]/dbname[?param1=value1&paramN=valueN]
+	// Find the last '/' (since the password or the net addr might contain a '/')
+	foundSlash := false
+	for i := len(dsn) - 1; i >= 0; i-- {
```

Comment:
Can we divide the loops into descriptive methods?

Reply:
This is an exact copy of the parser of the same library I'm trying to instrument (because each lib has it's own dsn mode), so I did not want to refactor it.

Reply:
In this case, the original code is here: https://github.com/go-sql-driver/mysql/blob/877a9775f06853f611fb2d4e817d92479242d1cd/dsn.go#L374-L449

Reply:
OK